runelite-client/src/main/java/net/runelite/client/ui/components/ToggleSwitch.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2018, Chickensalad <https://github.com/rubeng123>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.components;
+
+import net.runelite.client.plugins.config.ConfigPanel;
+
+import javax.imageio.ImageIO;
+import javax.swing.ImageIcon;
+import javax.swing.JLabel;
+import javax.swing.Timer;
+
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+
+public class ToggleSwitch extends JLabel implements ActionListener
+{
+
+	public static final int ON = 1;
+	public static final int OFF = 0;
+
+	private static final ImageIcon[] FRAME_ARRAY;
+	private static final int TOTAL_FRAMES = 5;
+
+	private int state = ON;
+
+	private int currentFrame = 4;
```

Comment:
Why does it start on 4?

Reply:
Following that, if your states go from
0 - off
1 - 25%
2 - 50%
3 - 75%
4 - on

then why do you have the state as an int and not a boolean, why use ints for that? and if you do use ints why isn't ON = 4?

Reply:
It starts on 4 because the 5th image in the array is the fully on frame, and the switch is initialized to the on position.

The state variable represents our current 'goal state' so we know which way we want to go when performing the animation. The value itself doesn't really mean anything, its just an alternative to creating an enum.

Reply:
Why not have an enum for this, you can still use ints if you call enum.ordinal(), this way you wouldn't need the on/off variables, and state could be of type ToggleState.

I'm not 100% sure this would be a better approach, @deathbeam thoughts?

Reply:
You're right, I should have used enum, not sure why I didn't in hindsight, I'll change it